lib/share_progress/variation.rb

```diff
@@ -0,0 +1,92 @@
+require 'share_progress'
+require 'share_progress/client'
+
+module ShareProgress
+  class Variation
+    class << self
+      def facebook_type_name
+        'facebook'
+      end
+
+      def twitter_type_name
+        'twitter'
+      end
+
+      def email_type_name
```

Comment:
The goal for these was to make it easy for the button to dispatch to the
correct variation class when pulling from the api
On Aug 25, 2015 4:43 PM, "NealJMD" <notifications@github.com> wrote:

> In lib/share_progress/variation.rb
> <https://github.com/SumOfUs/share_progress/pull/9#discussion_r37916830>:
>
> > @@ -0,0 +1,92 @@
> > +require 'share_progress'
> > +require 'share_progress/client'
> > +
> > +module ShareProgress
> > +  class Variation
> > +    class << self
> > +      def facebook_type_name
> > +        'facebook'
> > +      end
> > +
> > +      def twitter_type_name
> > +        'twitter'
> > +      end
> > +
> > +      def email_type_name
>
> what is this for? should these each be the type_name method of the
> respective classes?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SumOfUs/share_progress/pull/9/files#r37916830>.
>